src/Http/Routing/src/Matching/HostMatcherPolicy.cs

```diff
@@ -48,6 +62,93 @@ public bool AppliesToEndpoints(IReadOnlyList<Endpoint> endpoints)
             });
         }
 
+        public Task ApplyAsync(HttpContext httpContext, EndpointSelectorContext context, CandidateSet candidates)
```

Comment:
I didn't seem much reusable from the jump table implementation because the order of operations is different. It seemed more maintainable to just create a second code path that's about 85% separate.